File: android/app/src/main/java/com/example/whatsapp_plugin/models/Message.java
package com.example.whatsapp_plugin.models;

import java.util.Date;

public class Message {
    private String id;
    private String sender;
    private String text;
    private String groupName;
    private boolean isGroupMessage;
    private String date;



    public Message(String sender, String text, String date, boolean isGroupMessage, String groupName) {
        this.sender = sender;
        this.text = text;
        this.date = date;
        this.isGroupMessage = isGroupMessage;
        this.groupName = groupName;
    }

    public String getSender() {
        return sender;
    }

    public String getText() {
        return text;
    }

    public String getGroupName() {
        return groupName;
    }

    public String getDate() {
        return date;
    }

    public boolean isGroupMessage() {
        return isGroupMessage;
    }
}

File: lib/models/message.dart
import 'dart:convert';

class Message {
  final String groupName;
  final String id;
  final bool isGroupMessage;
  final String text;
  final String sender;

  Message(
      this.groupName,
      this.id,
      this.isGroupMessage,
      this.text,
      this.sender,
      );
  Message.fromJson(Map<String,dynamic> json)
    :groupName = json['groupName'],
     id = json['id'],
     isGroupMessage = json['isGroupMessage'],
     text = json['text'],
     sender = json['sender'];

  Map<String,dynamic> toJson() =>
      {
        'groupName' : groupName,
        'id' : id,
        'isGroupMessage' : isGroupMessage,
        'text' : text,
      };
}